### ISSUE TITLE
Parse Error: [ext_resource] referenced non-loaded resource when loading a specific scene

### ISSUE THREAD

**Issue by timshannon** [godotengine/godot#64330]
### Godot version

v4.0.alpha.mono.custom_build [67d401552]

### System information

Ubuntu 22.04

### Issue description

See the attached project for the scene in question.  You get this error when trying to load it:

```
  scene/resources/resource_format_text.cpp:177 - res://SlidingDoor/SlidingDoor.tscn:8 - Parse Error: [ext_resource] referenced non-loaded resource at: res://SlidingDoor/SlidingDoor.tscn
  scene/resources/resource_format_text.cpp:610 - res://SlidingDoor/SlidingDoor.tscn:8 - Parse Error: [ext_resource] referenced non-loaded resource at: res://SlidingDoor/SlidingDoor.tscn
  Failed loading resource: res://SlidingDoor/SlidingDoor.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  editor/editor_data.cpp:530 - Index p_idx = 1 is out of bounds (edited_scene.size() = 1).
```

### Steps to reproduce

Simply open the attached project.  The default scene fails to load.

When you re-import the blend file referenced in the `SlidingDoor.tscn` file the scene will load fine, until you save the scene which breaks it again.  You can keep going round and round like that.

I believe it has something to do with this particular blend file, but I have yet to figure out what specifically is causing the issue.   Also I had this issue on previous GD4 commits, but it went away and came back again, so I think a fix may have been reverted recently.  If I find the commit where it broke, I'll update this issue.

### Minimal reproduction project

[Bug.zip](https://github.com/godotengine/godot/files/9328584/Bug.zip)

**Comment by timshannon** [comment]
Same behavior happens in all alpha versions back through to alpha 9, so it maybe that this is a blender issue, as I've gone through a few version updates in blender since then as well.

Still investigating.

**Comment by timshannon** [comment]
I've removed everything from the blend file and replaced it with just a cube, and this issue still happens.

There is also no difference in the .tscn file when it can be opened and when it can't.  It's literally the same text:
```
[gd_scene load_steps=2 format=3 uid="uid://c7pdpypbs85g5"]

[ext_resource type="PackedScene" uid="uid://c7pdpypbs85g5" path="res://SlidingDoor/SlidingDoor.blend" id="1_n726j"]

[node name="SlidingDoor" instance=ExtResource("1_n726j")]
```

**Comment by timshannon** [comment]
Completely removed the .import and .godot folders, still same behavior.  Not sure what I'm missing.

**Comment by timshannon** [comment]
Also, FYI, the same behavior happens when you reopen the editor.  So the scene is fine until you restart the editor, then it's broken again.  So re-saving the scene or reopening the editor breaks it, and reimporting the blend file fixes it.

**Comment by rklassen** [comment]
having same issue

**Comment by tjw123hh** [comment]
having same issue

**Comment by Telsanyr** [comment]
Having the same issue here:

1. I was working on Godot, everything was fine, I could run my project, no error.
2. I did save my project in this state (git commit), before closing Godot
3. I closed Godot window
4. When i re-opened Godot, my GDScript was not attached to my scene anymore and had this error : 
> load: res://test-scenes/board/DebugTiles.tscn:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://test-scenes/board/DebugTiles.gd

**But**, when checking git, I see no modification since my last commit. The only files that I don't track with git are *.import files and .godot/ files

5. If I re-attach the script, the scene file is modified (git diff): 
> -[ext_resource type="Script" path="res://test-scenes/board/DebugTiles.gd" id="1_ys036"]
> +[ext_resource type="Script" path="res://test-scenes/board/DebugTiles.gd" id="1_bq1rw"]
>  
> [node name="DebugTiles" type="Node3D"]
> -script = ExtResource("1_ys036")
> +script = ExtResource("1_bq1rw")

If that can help.

**Comment by SimonPiCarter** [comment]
I have the exact same issue, for what it worth renaming the file to another name solved the issue.

**Comment by heppocogne** [comment]
@Telsanyr I had a similar issue and solved by using .scn instead of .tscn.

**Comment by Zireael07** [comment]
If renaming the file solves the problem, that would mean it's some sort of a cache problem...

**Comment by Necronomicron** [comment]
Just encountered this.
v4.1.dev2.official [668cf3c66]

**Comment by nongvantinh** [comment]
It happened to me today. I was feeling lazy to duplicate and rename things in the editor, so I opened the `Windows Explorer` and duplicated resources, renaming them there. Then, when I switched back to the editor, this error occurred.

Does anyone else edit resources outside the editor like I do? This is very likely one of the issues I had fixed for C# scripts in the past.

**Comment by njdf18** [comment]
I encountered this problem today. I have a node to transfer a player when it is touched. This node has an exported variable to let me drag a scene file into the inspector.
I noticed this error and reloading the editor, and all of my scenes became broken (error while loading even if there's no resource lost).
The solution I found is to edit your scene and remove all external scene references that are possible to cause the issue.

**Comment by chexmo** [comment]
I add up to @njdf18 with a similar use case... (using Godot 4.2.1.stable)
![image](https://github.com/godotengine/godot/assets/30608909/133730e4-d352-4ae7-bf0b-e8e30476795e)

**Comment by chexmo** [comment]
As an update... it seems like the error has something to be with the fact of coding several references in different scripts that points to the same scene... 

As a workaround, having a singleton that contains the references work
https://gitlab.com/rarodigma-studio/aleph/-/tree/PUBLIC/packed-scene-bug-workaround

However, it stills feel like a quick workaround... the benefit of using `@Export` is clearly better, I don't think it's a expected behaviour.

**Comment by pixel-boy** [comment]
I encountered the same problem ( using Godot 4.3.rc3 ), changing the typing of a variable from "ResourceCharacter" to "Resource" solved my problem (knowing that ResourceCharacter is a custom resource probably too referenced or something like that if I understand correctly)

**Comment by anonomity** [comment]
having same issue

**Comment by olivatooo** [comment]
Yeah, for me was because I used an Array[MyType]. After changing to Array it fixed this...
